agent/engine/docker_container_engine.go

```diff
@@ -726,7 +726,7 @@ func (dg *dockerGoClient) Stats(id string, ctx context.Context) (<-chan *docker.
 	go func() {
 		statsErr := client.Stats(options)
 		if statsErr != nil {
-			seelog.Warnf("Error retrieving stats for container %s: %v", id, statsErr)
+			seelog.Debugf("Error retrieving stats for container %s: %v", id, statsErr)
```

Comment:
Since the error handling is not done here anyway, having a WARN message here wouldn't help I think. Even when the container dies, you get a host of 'context canceled' or 'no such container' error messages here. I think it makes more sense for that logic to be with `container.processStatsStream`. Does that make sense?